pypeit/flatfield.py

```diff
@@ -520,14 +525,43 @@ def fit(self, debug=False):
 
             # Check for saturation of the flat. If there are not enough
             # pixels do not attempt a fit, and continue to the next
-            # slit.  TODO: set the threshold to a parameter?
+            # slit.
             good_frac = np.sum(onslit & (rawflat < nonlinear_counts))/np.sum(onslit)
+            # TODO: set the threshold to a parameter?
             if good_frac < 0.5:
-                # TODO: Used slit ID in these print statments instead of slit index
-                msgs.warn('Only {:4.2f}'.format(100*good_frac)
-                          + '% of the pixels on this slit are not saturated.' + msgs.newline()
-                          + 'Consider raising nonlinear_counts={:5.3f}'.format(nonlinear_counts) +
-                          msgs.newline() + 'Not attempting to flat field slit {:d}'.format(slit))
+                if tweak_slits:
```

Comment:
Might be clearer to change the error message from  "Only ... not saturated" to "More than ... are saturated."

Reply:
This was the original message.  Leaving it for now.